plugin/src/main/java/org/wildfly/swarm/plugin/maven/MavenArtifactResolvingHelper.java

```diff
@@ -54,15 +56,18 @@
 public class MavenArtifactResolvingHelper implements ArtifactResolvingHelper {
 
     public MavenArtifactResolvingHelper(ArtifactResolver resolver,
-                                        RepositorySystem system,
-                                        RepositorySystemSession session) {
+            RepositorySystem system,
+            RepositorySystemSession session) {
         this.resolver = resolver;
         this.system = system;
         this.session = session;
+        this.remoteRepositories.add(buildRemoteRepository("jboss-public-repository-group",
```

Comment:
Here is a summary of the changes: 
- Refactored the method by removing the session parameter (since it's an attribute) 
- Removed the duplicate call to `resolveArtifacts` and used `resolveDependencies` instead of `collectDependencies`
- Used `Objects::nonNull` instead of `x->x!= null`